cpp/src/aztec/stdlib/primitives/bool/bool.cpp

```diff
@@ -420,18 +420,68 @@ void bool_t<ComposerContext>::assert_equal(const bool_t& rhs, std::string const&
     }
 }
 
+// if predicate == true then return lhs, else return rhs
+template <typename ComposerContext>
+bool_t<ComposerContext> bool_t<ComposerContext>::conditional_assign(const bool_t<ComposerContext>& predicate,
```

Comment:
Its unused but I've added a test since we might use it:
https://github.com/AztecProtocol/barretenberg/blob/0414410d75b8e3e539e845b61c275071e53efe8b/cpp/src/barretenberg/stdlib/primitives/bool/bool.test.cpp#L472-L500